README.md

```diff
@@ -137,6 +168,77 @@ Can pass a single geojson object or an array of geojson objects
 FeatureServer.layers(geojson, options)
 ```
 
+### FeatureServer.generateRenderer
+Pass in `geojson` and `options`, and the function will return a valid generateRenderer object. Two `classificationDef` classification types are supported, _classBreaksDef_ and _uniqueValueDef_. 
+
+_classBreaksDef_ is used to classify numeric data based on a number of breaks and a statistical method. Features can also be normalized before being classified. _uniqueValueDef_ is used to classify data based on a unique field(s). If classification breaks are not supplied through in `statistics`, they will be generated using `classificationDef` options. The output is a _generateRenderer_ object.
+
+##### classBreaksDef
+In addition to class breaks as input `statistics: []`, the following is an example of _all_ classBreaksDef `options` that can be passed into the generateRenderer route: '/FeatureServer/:layer/generateRenderer'
+
+e.g.
+
+```js
+const options = {
+ *'classificationDef': {
+   *'type': 'classBreaksDef',
+   *'classificationField': '<field1>',
+   *'classificationMethod': 'esriClassifyEqualInterval' | 'esriClassifyNaturalBreaks' | 'esriClassifyQuantile' | 'esriClassifyStandardDeviation',
+   *'breakCount': 9,
+    'normalizationType': 'esriNormalizeByField' | 'esriNormalizeByLog' | 'esriNormalizeByPercentOfTotal',
+    'normalizationField': '<field2>' // mandatory if 'normalizationType' === 'esriNormalizeByField'
+    'baseSymbol': {
+      'type': 'esriSMS',
+      'style': 'esriSMSCircle',
+      'width': 2
+    },
+    'colorRamp': {
+      'type': 'algorithmic',
+      'fromColor': [115,76,0,255],
+      'toColor': [255,25,86,255],
+      'algorithm': 'esriHSVAlgorithm'
+    }
+  },
+  'where': '<field2> > 39'
+}
+
+FeatureServer.generateRender(geojson, options)
+
+*required
+```
+
+##### uniqueValueDef
+The following is an example of _all_ uniqueValueDef `options` that can be passed into the generateRenderer route: '/FeatureServer/:layer/generateRenderer'
+
+e.g.
+
+```js
+const options = {
```

Comment:
I agree, I don't want to bloat the docs, but `<details>` requires html, and `<code>` blocks don't do syntax highlighting. I am going to keep it as .md for now.